web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -260,7 +260,11 @@ private String getName(ServletContext context) {
         String name = null;
         try {
             String contextPath = context.getContextPath();
-            if (CommonUtils.isNullOrEmpty(contextPath)) {
+            TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.getActive();
+            if (telemetryConfiguration.getApplicationName() != null) {
+                name = telemetryConfiguration.getApplicationName();
```

Comment:
Do we need a check here if someone has tried setting application name as null?

Reply:
Thanks, I will add check into `TelemetryConfiguration#setApplicationName` that name is not null or empty.